sci-biology/abyss/abyss-2.0.2.ebuild

```diff
@@ -31,6 +31,10 @@ DEPEND="${RDEPEND}
 # todo: --enable-maxk=N configure option
 # todo: fix automagic mpi toggling
 
+pkg_setup() {
+	use openmp && ! tc-has-openmp && die "Please switch to an openmp compatible compiler"
```

Comment:
`use openmp && tc-check-openmp`
That's all you need (`tc-check-openmp` prints a standard uniform message)